src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -390,6 +390,54 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
         }
 
         public static bool CheckConstraints(
```

Comment:
A more straightforward approach would be to iterate through the types of the elements (`TupleElementTypes`) and check type constraints on each one. 
Then the diagnostics can trivially be associated with their corresponding element's position.